barter-data/src/exchange/bybit/book/l1.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    event::{MarketEvent, MarketIter},
+    exchange::{bybit::channel::BybitChannel, subscription::ExchangeSub, ExchangeId},
+    subscription::book::{Level, OrderBookL1},
+    Identifier,
+};
+use barter_integration::model::{Exchange, SubscriptionId};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize)]
+pub struct BybitOrderBookL1 {
+    pub topic: String,
+    #[serde(with = "chrono::serde::ts_milliseconds")]
+    pub ts: DateTime<Utc>,
+    #[serde(rename = "type")]
+    pub update_type: String,
+    pub data: BybitOrderBookL1Data,
+    pub cts: u64,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct BybitOrderBookL1Data {
+    pub s: String,
+    pub b: Vec<[String; 2]>,
+    pub a: Vec<[String; 2]>,
```

Comment:
these should be deserialized into number types here not later on

see the coinbase PR for an example:
```rust
    #[serde(
        rename(deserialize = "best_bid_size"),
        deserialize_with = "barter_integration::de::de_str"
    )]
    pub best_bid_amount: f64,
```

actually has this lying around
```rust
/// https://bybit-exchange.github.io/docs/v5/websocket/public/orderbook#response-example
pub fn de_ob_l1_level<'de, D>(deserializer: D) -> Result<[f64; 2], D::Error>
where
    D: serde::de::Deserializer<'de>,
{
    let mut value = [0.0, 0.0];
    if let Ok(level) = <[[&str; 2]; 1] as Deserialize>::deserialize(deserializer) {
        value[0] = level[0][0].parse().map_err(serde::de::Error::custom)?;
        value[1] = level[0][1].parse().map_err(serde::de::Error::custom)?;
    }

    Ok(value)

}
```

Reply:
O nice yes that is a lot cleaner 💯 

also does this mean it'll return [0.0,0.0] sometimes on error?

Reply:
ah saw u posted about this in the slack lol